src/generate/azgenerator.ts

```diff
@@ -47,7 +53,23 @@ export async function processRequest(host: Host) {
 
         for (const f in files) {
             if (!isNullOrUndefined(files[f])) {
-                host.WriteFile(f, files[f].join(EOL));
+                if (
+                    (AzConfiguration.getValue(CodeGenConstants.generationMode) !=
```

Comment:
Can you simplify this logic, there are so many conditions here.

Reply:
Currently we have only supported the following files
```
azext_metadata.json
setup.py
HISTORY.rst
tests/__init__.py
preparers.py
```
 to be generated with code templates. and the `azext_metadata.json` and `setup.py` has another special logic in incremental generation mode. This can't be done with code templates. 
And this condition here 
```
(f.endsWith('setup.py') &&
     AzConfiguration.getValue(CodeGenConstants.targetMode) !==
             TargetMode.Core)
```
 is because main repo generation will also have a setup.py to be modified, we can't use code templates either.